src/content/blog/shorebird-code-push-v2.md

```diff
@@ -0,0 +1,161 @@
+---
+title: What's New in v2.0 of the Shorebird Code Push Package
+author: felangel
+description: Announcing the pre-release of pkg:shorebird_code_push v2.0
+date: 2024-11-04
+cover: /blog/shorebird-code-push-v2-cover.png
+coverAlt: Improved Patch Delivery Cover Image
+---
+
+We're excited to announce a pre-release of version 2.0 of the
+[`shorebird_code_push`
+package](https://pub.dev/packages/shorebird_code_push/versions/2.0.0-dev.2).
+
+## What's New
+
+We've taken some time to re-imagine what the Dart API should look like when
+interfacing with the Shorebird updater and we've re-written the Dart API from the
+ground up.
+
+The two main areas we focused on were:
+
+- ✨ Simplifying the public API
+- ❗️ Improving error reporting/handling
+
+We really wanted `package:shorebird_code_push` to be simple and intuitive to use
+so v2 comes with a much smaller public API surface.
+
+The first thing we did was rename the core class from `ShorebirdCodePush` to
+`ShorebirdUpdater` in order to better describe the purpose of the class.
+
+```dart
+import 'package:shorebird_code_push/shorebird_code_push.dart';
+
+// Create an instance of the `ShorebirdUpdater`
+final updater = ShorebirdUpdater();
+```
+
+We broke the public API into 3 main parts:
+
+1. Reading patch information
+1. Checking for updates
+1. Performing updates
+
+### Reading Patch Information
+
+To get information about the current (installed) or next (downloaded) patches,
+you can use the `readCurrentPatch` or `readNextPatch` APIs.
+
+```dart
+@override
+void initState() {
+  super.initState();
+  updater.readCurrentPatch().then((patch) {
+    // The current patch is `null` when no patch is installed.
+    print(patch?.number);
+  });
+}
+```
+
+### Checking for Updates
+
+To check for updates, you can use the `checkForUpdate` API:
+
+```dart
+Future<void> _checkForUpdate() async {
+  // Check if there's an update available.
+  final status = await updater.checkForUpdate();
+  if (!mounted) return;
```

Comment:
This is only if this function is within a widget?

Reply:
Yeah all code snippets are assuming the context is within a stateful widget (e.g. the `initState` above)

Reply:
I simplified the snippets